Source/Core/VideoBackends/D3D/VertexShaderCache.cpp

```diff
@@ -159,18 +159,29 @@ void VertexShaderCache::Init()
   SETSTAT(stats.numVertexShadersAlive, 0);
 
   if (g_ActiveConfig.bShaderCache)
-  {
-    if (!File::Exists(File::GetUserPath(D_SHADERCACHE_IDX)))
-      File::CreateDir(File::GetUserPath(D_SHADERCACHE_IDX));
-
-    std::string cache_filename =
-        StringFromFormat("%sdx11-%s-vs.cache", File::GetUserPath(D_SHADERCACHE_IDX).c_str(),
-                         SConfig::GetInstance().GetGameID().c_str());
-    VertexShaderCacheInserter inserter;
-    g_vs_disk_cache.OpenAndRead(cache_filename, inserter);
-  }
+    LoadShaderCache();
+}
 
-  last_entry = nullptr;
+void VertexShaderCache::LoadShaderCache()
```

Comment:
Could some or all of this be moved to common?  I see the filename generated very similarly between all 3 backends.  They all do something similar with the OpenAndRead on the disk cache (though Vulkan does a lot more)